contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -500,16 +507,39 @@ resolve_numeric_typmod_from_exp(Node *expr)
 			 * Refer to details of precision and scale calculation in the following link:
 			 * https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/data-types/precision-scale-and-length-transact-sql.md
 			 */
+			has_aggregate_operand = arg1->type == T_Aggref ||
+						(list_length(op->args) == 2 && arg2->type == T_Aggref);
+
 			switch (op->opfuncid)
 			{
 				case NUMERIC_ADD_OID:
 				case NUMERIC_SUB_OID:
+					integralDigitCount = Max(precision1 - scale1, precision2 - scale2);
 					scale = Max(scale1, scale2);
-					precision = Max(precision1 - scale1, precision2 - scale2) + 1 + scale;
+					precision = integralDigitCount + 1 + scale;
+					/*
+					 * For addition and subtraction, skip scale adjustment when none of the
+					 * operands is part of any aggregate function
+					 */
+					if (has_aggregate_operand &&
+					    integralDigitCount < (Min(TDS_MAX_NUM_PRECISION, precision) - scale))
```

Comment:
can you explain with concrete example. After reading doc it seems we reduce scale to make sure that integral part is not truncated but here we seem to increase the scale. Take this example 
pr1 = 35 sc1 = 5
pr2 = 35 sc1 = 5

integDC = 30
scale = 5
pr = 30 + 1 + 5 = 36

now 30 < 36 - 5
   scale = 36 - 30 = 6 (why are we doing this?)

Reply:
For the above example, resultant pr = 36, resultant scale = 6  we are not really losing any data on integral part(i.e., precision - scale) we have the same 30 digits; we are appending an extra zero in decimal part (right most digit in scale) which is again a SQL server behaviour.

Reply:
what I understand is that this helps in making sure that we do not truncate integer part by adjusting scale part. Can you give an example where this code will achieve that?